ml/curatedBreastData-InfoGAN/funs_common.py

```diff
@@ -0,0 +1,120 @@
+    "study_17705_GPL96_MDACC_Tissue_BC_Tamoxifen-bmc15.csv.xz",
+    "study_19615_GPL570_all-bmc15.csv.xz",
+    "study_20181_GPL96_all-bmc15.csv.xz",
+    "study_20194_GPL96_all-bmc15.csv.xz",
+    "study_2034_GPL96_all-bmc15.csv.xz",
+    "study_22226_GPL1708_all-bmc15.csv.xz",
+    "study_22358_GPL5325_all-bmc15.csv.xz",
+    "study_25055_GPL96_MDACC_M-bmc15.csv.xz",
+    "study_25065_GPL96_MDACC-bmc15.csv.xz",
+    "study_25065_GPL96_USO-bmc15.csv.xz",
+    "study_32646_GPL570_all-bmc15.csv.xz",
+    "study_9893_GPL5049_all-bmc15.csv.xz"]
+
+    datasets = [pd.read_csv(os.path.join(path,d)) for d in datasets_fn]
+    join_dataset = pd.concat(datasets, sort=False, ignore_index=True)
+    
+    treat_dataset = read_treat_dataset()
+    return pd.merge(treat_dataset, join_dataset)
+
+def read_mike_dataset():
+    join_dataset = pd.read_csv("ex15bmcMerged.csv")
+    treat_dataset = read_treat_dataset()
+    return pd.merge(treat_dataset, join_dataset)
+
+def read_combat_dataset():
+    #join_dataset = pd.read_csv("merged-combat15.csv")
+    join_dataset = pd.read_csv("data/ex15bmcMerged.csv.xz")
+    treat_dataset = read_treat_dataset()
+    return pd.merge(treat_dataset, join_dataset)
+    
+
+def drop_trea(full_dataset):
+    return full_dataset.drop(columns=['radio','surgery','chemo','hormone'])
+
+def get_pam50_list():
+    return ['ACTR3B', 'ANLN', 'BAG1', 'BCL2', 'BIRC5', 'BLVRA', 'CCNB1', 'CCNE1', 'CDC20', 'CDC6', 'NUF2', 'CDH3', 'CENPF', 'CEP55', 'CXXC5', 'EGFR', 'ERBB2', 'ESR1', 'EXO1', 'FGFR4', 'FOXA1', 'FOXC1', 'GPR160', 'GRB7', 'KIF2C', 'NDC80', 'KRT14', 'KRT17', 'KRT5', 'MAPT', 'MDM2', 'MELK', 'MIA', 'MKI67', 'MLPH', 'MMP11', 'MYBL2', 'MYC', 'NAT1', 'ORC6', 'PGR', 'PHGDH', 'PTTG1', 'RRM2', 'SFRP1', 'SLC39A6', 'TMEM45B', 'TYMS', 'UBE2C', 'UBE2T']
+    
+def select_pam50(full_dataset):
+    pam50 = get_pam50_list()
+    selected_pam50 = list(set(pam50).intersection(set(full_dataset)))
+    return full_dataset[selected_pam50 + ['study', 'posOutcome', 'patient_ID', 'pCR', 'RFS', 'DFS']]
+    
+def prepare_full_dataset(full_dataset, y_field = 'posOutcome'):
+    X = full_dataset.drop(columns=['study', 'patient_ID','pCR', 'RFS', 'DFS', 'posOutcome']).to_numpy()
+    y_posOutcome = full_dataset[y_field].to_numpy()
+    return X, y_posOutcome
+
+
+def prepare_dataset(full_dataset, study, y_field = 'posOutcome'):
+    study_dataset = full_dataset.loc[full_dataset['study'] == study]
+    X = study_dataset.drop(columns=['study', 'patient_ID','pCR', 'RFS', 'DFS', 'posOutcome']).to_numpy(dtype = np.float)
+    y_posOutcome = study_dataset[y_field].to_numpy(dtype = np.float)
+    return X, y_posOutcome
+
+def prepare_datasets(full_dataset, studies):
+    Xys = [prepare_dataset(full_dataset, study ) for study in studies]
+    Xs, ys = zip(*Xys)
+    return np.concatenate(Xs), np.concatenate(ys)
+            
+            
+def list_to_4g_str(l):
+    return "  ".join("%.4g"%d for d in l)
+    
+def list2d_to_4g_str_pm(l):
```

Comment:
Where this function is used?

Reply:
In experement2.6_combat_pcr_rfs_dfs.py
list2d_to_4g_str_pm deleted